src/common/arrow/arrow_appender.cpp

```diff
@@ -334,16 +341,20 @@ struct ArrowVarcharData {
 		auto validity_data = (uint8_t *)append_data.validity.data();
 
 		// resize the offset buffer - the offset buffer holds the offsets into the child array
-		append_data.main_buffer.resize(append_data.main_buffer.size() + sizeof(uint64_t) * (size + 1));
+		append_data.main_buffer.resize(append_data.main_buffer.size() + sizeof(BUFTYPE) * (size + 1));
 		auto data = (SRC *)format.data;
-		auto offset_data = (uint64_t *)append_data.main_buffer.data();
+		auto offset_data = (BUFTYPE *)append_data.main_buffer.data();
 		if (append_data.row_count == 0) {
 			// first entry
 			offset_data[0] = 0;
 		}
 		// now append the string data to the auxiliary buffer
 		// the auxiliary buffer's length depends on the string lengths, so we resize as required
 		auto last_offset = offset_data[append_data.row_count];
+		idx_t max_offset = append_data.row_count + to - from;
+		if (max_offset > NumericLimits<uint32_t>::Maximum() && !append_data.export_large) {
+			throw InvalidInputException("Arrow Appender: Buffer type is not sufficiently big");
```

Comment:
Could this error message be expanded somewhat - e.g. `"the maximum total string size for regular string buffers is %X but the offset of %Y exceeds this."`